atlassian/rest_client.py

```diff
@@ -480,16 +715,69 @@ def patch(
             return response
         return self._response_handler(response)
 
+    # advanced False
+    @overload
     def delete(
         self,
-        path,
-        data=None,
-        headers=None,
-        params=None,
-        trailing=None,
-        absolute=False,
-        advanced_mode=False,
-    ):
+        path: str,
+        data: dict | str | None = ...,
+        headers: dict | None = ...,
+        params: dict | None = ...,
+        trailing: bool | None = ...,
+        absolute: bool = ...,
+        *,
+        advanced_mode: Literal[False],
+    ) -> T_resp_json: ...
+
+    @overload
+    def delete(
+        self,
+        path: str,
+        data: dict | str | None = ...,
+        headers: dict | None = ...,
+        params: dict | None = ...,
+        trailing: bool | None = ...,
+        absolute: bool = ...,
+        advanced_mode: Literal[False] = ...,
+    ) -> T_resp_json: ...
+
+    # advanced True
+    @overload
+    def delete(
+        self,
+        path: str,
+        data: dict | str | None = ...,
+        headers: dict | None = ...,
+        params: dict | None = ...,
+        trailing: bool | None = ...,
+        absolute: bool = ...,
+        *,
+        advanced_mode: Literal[True],
+    ) -> Response: ...
+
+    # basic overall case
+    @overload
+    def delete(
+        self,
+        path: str,
+        data: dict | str | None = ...,
+        headers: dict | None = ...,
+        params: dict | None = ...,
+        trailing: bool | None = ...,
+        absolute: bool = ...,
+        advanced_mode: bool = ...,
+    ) -> T_resp: ...
+
+    def delete(
+        self,
+        path: str,
+        data: dict | str | None = None,
+        headers: dict | None = None,
+        params: dict | None = None,
+        trailing: bool | None = None,
+        absolute: bool = False,
+        advanced_mode: bool = False,
+    ) -> T_resp:
```

Comment:
For those unfamiliar with how overloads work this can look confusing.  Essentially this allows mypy to understand that certain arguments change the output type of the method.  Due to some of the of the calls using positional arguments forces us to include additional overrides.   I wish there was an easier way to make this work with mypy.